penguin/engine.py

```
@@ -1904,268 +1969,68 @@ async def run_task(
-            if message_callback:
-                await message_callback(f"LLM Empty Response: {str(e)}", "error")
+            cm, _api, _tm, _ae = self._resolve_components(self.current_agent_id)
+            cm.conversation.prepare_conversation(task_prompt, image_paths=image_paths)
+
+            telemetry = getattr(self, "telemetry", None)
+            if telemetry is not None:
+                await telemetry.record_task(self.current_agent_id, task_name)
+
+            task_metadata = {
+                "id": task_id
+                or f"task_{datetime.utcnow().strftime('%Y%m%d_%H%M%S')}_{uuid.uuid4().hex}",
```

Ruff (0.15.10)

[warning] `datetime.datetime.utcnow()` used

(DTZ003)